src/devices/video/hd44780.cpp

```diff
@@ -108,6 +147,8 @@ const tiny_rom_entry *hd44780_device::device_rom_region() const
 	switch (m_charset_type)
 	{
 		case CHARSET_HD44780_A00:   return ROM_NAME( hd44780_a00 );
+		case CHARSET_HD44780U_A00:  return ROM_NAME( hd44780u_a00 );
+		case CHARSET_HD44780U_A02:  return ROM_NAME( hd44780u_a02 );
 		case CHARSET_SED1278_0B:    return ROM_NAME( sed1278_0b );
 		case CHARSET_KS0066_F00:    return ROM_NAME( ks0066_f00 );
 		case CHARSET_KS0066_F05:    return ROM_NAME( ks0066_f05 );
```

Comment:
This is really gross.  Why aren’t you overriding `device_rom_region` in each device with a different ROM?

Reply:
I followed the same coding pattern/style that the device already had.

Reply:
Doesn’t change the fact that it’s completely nonsensical, and it should be pretty obvious there are no other devices doing this.

Reply:
Isn't Sandro the correct person to direct that to, not LN?

Reply:
Well in a perfect world, we could go back and ensure it didn’t get to this state over a decade ago.  But the way it got into this state was that there was originally a single device type with the ROM selected after it was instantiated at machine configuration time.  You couldn’t set device BIOS options from machine configuration back then, so there wasn’t a proper way to select between multiple ROM choices for a single device type, and this is what we ended up with.  Of course, back then devices didn’t participate in `-validate`, `-romident`, etc. so hackery like that wasn’t as visible to end users.

Anyway, we’re carrying so many of these things that are in a bad state due to accidents of history that we really need to maintain a policy of, “If you encounter something nonsensical, don’t perpetuate it.”